unit-tests/run-unit-tests.py

```diff
@@ -204,10 +204,25 @@ def check_log_for_fails( path_to_log, testname, configuration = None ):
 
 
 def get_tests():
-    global regex, target, pyrs, current_dir, linux
+    global regex, target, pyrs, current_dir, linux, context, list_only
     if regex:
         pattern = re.compile( regex )
-    if target:
+    if list_only:
+        # We want to list all tests, even if they weren't built.
+        # So we look for the source files instead of using the manifest
+        for cpp_test in file.find( current_dir, '(^|/)test-.*\.cpp' ):
+            testparent = os.path.dirname( cpp_test )  # "log/internal" <-  "log/internal/test-all.py"
+            if testparent:
+                testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename( cpp_test )[
+                                                                            5:-4]  # remove .cpp
+            else:
+                testname = os.path.basename( cpp_test )[:-4]
+
+            if regex and not pattern.search( testname ):
+                continue
+
+            yield libci.ExeTest( testname )

```

Comment:
Hmm this might still have an .exe and it's a shame it not complete, but I guess it's good enough for now...